src/Api/Controllers/OrganizationsController.cs

```diff
@@ -599,5 +605,53 @@ public async Task<OrganizationKeysResponseModel> PostKeys(string id, [FromBody]O
             var org = await _organizationService.UpdateOrganizationKeysAsync(new Guid(id), model.PublicKey, model.EncryptedPrivateKey);
             return new OrganizationKeysResponseModel(org);
         }
+
+        [HttpGet("{id:guid}/sso")]
+        public async Task<OrganizationSsoResponseModel> GetSso(Guid id)
+        {
+            if (!await _currentContext.ManageSso(id))
+            {
+                throw new NotFoundException();
```

Comment:
Should we use `UnauthorizedAccessException` instead?

Reply:
We usually use `NotFoundException` when the user lacks permission. I'm not sure if I completely agree with the pattern, but we should follow the same pattern everywhere.

Reply:
Alright, that seems odd to me as well, but if that's the pattern then let's follow it.